### PR TITLE
Documentation for the Controllers folder + minor detail for previous docstrings

### PR DESCRIPTION
**Pull request by jgrguric96**
This PR primarily contains the addition of docstrings for the files within the Controllers folder.
It also contains some small additions for the Agents folder files (such as type hints)

Luckily, a lot of context needed for the documentation was already done by mosaik itself with the requirements they have when inheriting the mosaik_api.Simulator class.

The missing docstring details can be expanded upon at a later date.

### CODE REVIEW

File: Agents/prosumer_mosaik.py [hunk 1]
```diff
@@ -41,16 +57,16 @@ def init(self, sid:str, time_resolution:float, step_size:int=900) -> dict:
...
         ----------
         sid : str
-            The ID in string format
+            The String ID of the class
         time_resolution : float
-            Sets the self.time_resolution variable
+            ???
         step_size : int
-            Sets the self.step_size variable
+            The size of the time step. The unit is arbitrary, bit it has to be consistent among all simulators used in a simulation.
```
Comment (jgrguric96) [inline Agents/prosumer_mosaik.py:64]: I already see a typo I made 👎

File: Controllers/ResidentialController/controller_mosaik.py [hunk 2]
```diff
@@ -40,7 +40,27 @@
 
 
 class controlSim(mosaik_api.Simulator):
-    def __init__(self):
+    def __init__(self) -> None:
+        """
+        Constructor for the controlSim class with the mosaik_api.Simulator class as its Parent class
```
Comment (manuGil) [inline Controllers/ResidentialController/controller_mosaik.py:45]: Read my comment on redundancy in the doc strings, and correct things across files

File: Games/p2ptrading_mosaik.py [hunk 3]
```diff
@@ -24,13 +24,49 @@
...
         super().__init__(META)
         self.eid_prefix = 'p2ptrading_'
         self.entities = {}
         self._cache = {}  #
 
-    def init(self, sid, time_resolution, step_size=900):
+    def init(self, sid:str, time_resolution:float, step_size:int=900) -> dict:
+        """
+        Initialize the simulator with the ID `sid` and pass the `time_resolution` and additional parameters sent by mosaik.
+        Because this method has an additional parameter `step_size` it is overriding the parent method init().
```
Comment (manuGil) [inline Games/p2ptrading_mosaik.py:52]: overwriting?

**Review by manuGil — CHANGES_REQUESTED** [review]
Thanks for writing the doc strings. Please read my comment on specific files. 
After that you can merge the changes. As a follow up, we should ask researchers to review and complete the missing information.

**Pull request merged**